src/pages/sidepanel/index.tsx

```diff
@@ -8,18 +10,64 @@ function init() {
   if (!appContainer) {
     throw new Error('Can not find #app-container');
   }
+
+  ReactDOM.render(<IndexSidepanel />, appContainer);
 }
 
 init();
 
-chrome.runtime.sendMessage({ greeting: 'Hello' }, function (response) {
-  console.log(response.farewell);
-});
-
-chrome.runtime.onMessage.addListener((message, _sender, sendResponse) => {
-  if (message.action === 'getContent') {
-      const content = 'This is the content from the background script';
-      sendResponse({content: content});
+chrome.runtime.onMessage.addListener(
+  function(request, sender, sendResponse) {
+    console.log(sender.tab ? "from a content script:" + sender.tab.url : "from the extension");
+    if (request.greeting === "hello") sendResponse({farewell: "goodbye"});
   }
-  return true;
-});
+);
+
+function IndexSidepanel(): JSX.Element {
+  const [currentUrl, setCurrentUrl] = useState<string>("");
+  const [formData, setFormData] = useState<{ name: string, email: string }>({ name: '', email: '' });
```

Comment:
```ts
type FormField = {
  name: string | null,
  type: string,
  id: string | null,
  value: string,
 }
```
The onMessage handler should get array of the FormField and display it in the HTML as just text for now.